ddtrace/contrib/botocore/patch.py

```diff
@@ -99,6 +56,8 @@ def patched_api_call(original_func, instance, args, kwargs):
 
             if endpoint_name == 'lambda' and operation == 'Invoke':
                 inject_trace_to_client_context(args, span)
+            if endpoint_name == 'sqs' and operation == 'SendMessage':
+                inject_trace_to_message_metadata(args, span)
```

Comment:
So something we didn't think about with the lambda injection that's worth considering now:

These operations are pretty expensive with all the encoding and decoding that is happening. Right now this is happening by default and cannot be disabled. I'm thinking that we should offer a configuration option to opt-out of this for users that might not be doing distributed tracing to their serverless apps. What are your thoughts @Czechh?

Reply:
This makes sense to me I think it's a good experience for users, but this of course has the caveat that there is an additional configuration that we'll need to support for ever. Is there an example of some other feature that we expose in a config?

Reply:
BTW in the SQS case we don't need to decode/encode.

Reply:
Ah good point about SQS. Okay let's leave it for a follow-up PR then 🙂. We would do it similarly to Celery for distributed tracing as introduced in https://github.com/DataDog/dd-trace-py/pull/1194

Reply:
Ok I'll add a following PR for AWS Lambda. Should we disable it by default?